kubeadm/v1.15.0/KubeClusterHelper.psm1

```diff
@@ -0,0 +1,1521 @@
+        clusterCIDR = $ClusterCIDR;
+        clientConnection = @{
+            kubeconfig = $KubeConfig
+        };
+        winkernel = @{
+            enableDSR = ($ProxyFeatureGates -match "WinDSR=true");
+            networkName = $NetworkName;
+        };
+    }
+
+    if ($ProxyFeatureGates -match "WinDSR=true")
+    {
+        $proxyArgs +=  "--enable-dsr=true" # Comment for config
+    }
+
+    if ($SourceVip)
+    {
+        $proxyArgs +=  "--source-vip=$SourceVip" # Comment out for config
+
+        $KubeproxyConfiguration.winkernel += @{
+            sourceVip = $SourceVip;
+        }
+    }
+    ConvertTo-Json -Depth 10 $KubeproxyConfiguration | Out-File -FilePath $KubeProxyConfig
+    #$proxyArgs += "--config=$KubeProxyConfig" # UnComment for Config
+    
+    return $proxyArgs
+}
+
+function InstallKubelet()
+{
+    param
+    (
+        [parameter(Mandatory=$true)] [string] $CniDir,
+        [parameter(Mandatory=$true)] [string] $CniConf,
+        [parameter(Mandatory=$true)] [string] $KubeDnsServiceIp,
+        [parameter(Mandatory=$true)] [string] $NodeIp,
+        [parameter(Mandatory = $false)] $KubeletFeatureGates = ""
+    )
+
+    Write-Host "Installing Kubelet Service"
+    $logDir = [io.Path]::Combine($(GetLogDir), "kubelet")
+    CreateDirectory $logDir 
+
+    $kubeletArgs = GetKubeletArguments -CniDir $CniDir `
+                    -CniConf $CniConf   `
+                    -KubeDnsServiceIp $KubeDnsServiceIp `
+                    -NodeIp $NodeIp -KubeletFeatureGates $KubeletFeatureGates `
+                    -LogDir $logDir
+
+    #CreateService -ServiceName Kubelet -CommandLine $kubeletArgs -LogFile "$log"
+
+    $kubeletBinPath = $((get-command kubelet.exe -ErrorAction Stop).Source)
+
+    #New-Service -Name "kubelet" -StartupType Automatic -BinaryPathName "$kubeletBinPath --windows-service --v=6 --log-dir=$logDir --cert-dir=$env:SYSTEMDRIVE\var\lib\kubelet\pki --cni-bin-dir=$CniDir --cni-conf-dir=$CniConf --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf --hostname-override=$hostname --pod-infra-container-image=$Global:PauseImage  --allow-privileged=true --enable-debugging-handlers  --cgroups-per-qos=false  --logtostderr=false  --network-plugin=cni --enforce-node-allocatable=`"`" --feature-gates=$KubeletFeatureGates"
+    New-Service -Name "kubelet" -StartupType Automatic -BinaryPathName "$kubeletArgs"
+    kubeadm join "$(GetAPIServerEndpoint)" --token "$Global:Token" --discovery-token-ca-cert-hash "$Global:CAHash"
+    # Open firewall for 10250. Required for kubectl exec pod <>
+    if (!(Get-NetFirewallRule -Name KubeletAllow10250 -ErrorAction SilentlyContinue ))
+    {
+        New-NetFirewallRule -Name KubeletAllow10250 -Description "Kubelet Allow 10250" -Action Allow -LocalPort 10250 -Enabled True -DisplayName "KubeletAllow10250" -Protocol TCP -ErrorAction Stop
+    }
+    if (!(IsNodeRegistered)) {
+        Write-Host "$(IsNodeRegistered)"
```

Comment:
isn't this just going to `Write-Host $false`?

Reply:
Removed